### PR TITLE
Switch to modules for epic

### PR DESCRIPTION
**Pull request by nicl**
**DO NOT MERGE:** depends on https://github.com/guardian/contributions-service/pull/194.

## What does this change?

This includes switching to the v2 of automat client libraries, and loading the epic as an ES module.

See also:

https://github.com/guardian/contributions-service/pull/194

https://github.com/guardian/automat-client-v2/

https://docs.google.com/document/d/1CVkWvCo5SlJjedPmv0Uh6gQ6mwFsGydE2RvZzo6uxGk/edit?usp=sharing

## Screenshots

Screenshot 2020-07-23 at 16 43 49

edit: update because touches banner code too:

Screenshot 2020-07-27 at 16 28 58

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/contributions-service.js [hunk 1]
```diff
@@ -180,7 +108,8 @@ const buildEpicPayload = () => {
         mvtId: getMvtValue(),
         countryCode,
         epicViewLog: getViewLog(),
-        weeklyArticleHistory: getWeeklyArticleHistory()
+        weeklyArticleHistory: getWeeklyArticleHistory(),
+        hasOptedOutOfArticleCount: false,
```
Comment (tomrf1) [inline static/src/javascripts/projects/common/modules/commercial/contributions-service.js:112]:
```
import {ARTICLES_VIEWED_OPT_OUT_COOKIE} from 'common/modules/commercial/user-features';
...
hasOptedOutOfArticleCount: getCookie(ARTICLES_VIEWED_OPT_OUT_COOKIE.name)
```

File: static/src/javascripts/projects/common/modules/commercial/contributions-service.js [hunk 2]
```diff
@@ -317,43 +241,48 @@ export const renderBanner: (BannerDataResponse) => Promise<boolean> = (response)
...
-                            meta.abTestVariant.tickerSettings,
-                        )})
-            }
-        })
-        .catch(error => {
-            console.log(error);
-            reportError(error, {}, false);
-        });
-};
+        const response = await getEpicMeta(payload, "http://localhost:3030/epic");
```
Comment (tjmw) [inline static/src/javascripts/projects/common/modules/commercial/contributions-service.js:252]: Reminder to update this host!

**Review by tjmw — APPROVED** [review]
Nice - this looks really great 🏅

So does this new approach solve the two outstanding issues we'd seen with remote modules on frontend (issues with src-button and issues with Shadow DOM & Emotion)?

File: static/src/javascripts/projects/common/modules/commercial/contributions-service.js (hunk 2 above)
Comment (nicl) [reply static/src/javascripts/projects/common/modules/commercial/contributions-service.js:252]: ah thanks, I always seem to do this :(

**Comment by nicl** [comment]
@tjmw yes, this solves the shadow-dom/emotion issues!

File: package.json [hunk 3]
```diff
@@ -14,8 +14,10 @@
     "@emotion/core": "^10.0.27",
     "@emotion/styled": "^10.0.27",
     "@guardian/atom-renderer": "1.1.6",
-    "@guardian/automat-client": "^0.2.16",
     "@guardian/consent-management-platform": "4.0.0-8",
+    "@guardian/automat-contributions": "^0.3.5",
+    "@guardian/automat-modules": "^0.3.3",
+    "@guardian/consent-management-platform": "4.0.0-5",
```
Comment (mxdvl) [inline package.json:20]: CMP v4.0.0-5 and -8 are _pre-release_ versions, are you sure it is correct that they are pointing to these?

Comment (nicl) [reply package.json:20]: Hey, it's probably a mistake in my rebase against local master. Will bump.

Comment (mxdvl) [reply package.json]: No worries!

**Comment by jfsoul** [comment]
@nicl do we know why `automat-deps` removes the problem we had with `src-button`? IIRC we tried aligning all the versions, but I might be wrong.

**Comment by nicl** [comment]
@jfsoul `@guardian/automat-deps` isn't really the solution here in fact, though aligning versions should protect against future issues caused by version mismatch. 

I think there were two problems:

**1)** styles weren't appearing with shadow-dom fullstop

e.g. see: https://github.com/guardian/frontend/pull/22738.

This is fixed by using @emotion/core for everything rather than emotion. As you can't use a cache provider (required for [container](https://emotion.sh/docs/@emotion/cache#container)) for vanilla emotion.

**2)** even without shadow-dom, source styles weren't working

I can't remember on this to be honest why this was happening.

**Pull request merged**